pallet-chainlink-feed/src/lib.rs

```diff
@@ -730,15 +767,22 @@ pub mod pallet {
 
 				// update oracle rewards and try to reserve them
 				let payment = details.payment;
-				T::Currency::reserve(&T::PalletId::get().into_account(), payment).or_else(
+				// track the debt in case we cannot reserve
+				T::Currency::reserve(&Self::account_id(), payment).or_else(
 					|_| -> DispatchResult {
 						// track the debt in case we cannot reserve
-						Debt::<T>::try_mutate(|debt| {
-							*debt = debt.checked_add(&payment).ok_or(Error::<T>::Overflow)?;
-							Ok(())
-						})
+						let mut new_debt = feed.config.debt.clone();
```

Comment:
Isn't Balance also `Copy`?

Reply:
ur right, old habits of writing Javascript LOL